Fails to build: defined ‘static’ in ‘constexpr’ function only available with ‘-std=c++23’

### What is the issue?

4.1.0-beta.4 fails to build with the following error:
```
libtransmission/variant.h:104:35: error: ‘Predicate’ defined ‘static’ in ‘constexpr’ function only available with ‘-std=c++23’ or ‘-std=gnu++23’
  104 |             static auto constexpr Predicate = [](auto const& item, tr_quark key)
      |                                   ^~~~~~~~~
```
This patch works around the problem, not sure it is the right fix. Maybe the function should be declared `TR_CONSTEXPR22` instead of `TR_CONSTEXPR20`.
```diff
--- transmission.git.orig/libtransmission/variant.h     2025-12-05 10:22:14.853134119 +0100
+++ transmission.git/libtransmission/variant.h  2025-12-05 10:23:02.262034462 +0100
@@ -101,7 +101,7 @@

         [[nodiscard]] TR_CONSTEXPR20 auto find(std::initializer_list<tr_quark> keys) noexcept
         {
-            static auto constexpr Predicate = [](auto const& item, tr_quark key)
+            auto constexpr Predicate = [](auto const& item, tr_quark key)
             {
                 return item.first == key;
             };
```

### Which application of Transmission?

None

### Which version of Transmission?

_No response_

--- a/libtransmission/variant.h
+++ b/libtransmission/variant.h
@@ -101,7 +101,7 @@ public:
 
         [[nodiscard]] TR_CONSTEXPR20 auto find(std::initializer_list<tr_quark> keys) noexcept
         {
-            static auto constexpr Predicate = [](auto const& item, tr_quark key)
+            auto constexpr Predicate = [](auto const& item, tr_quark key)
             {
                 return item.first == key;
             };

